rules/heading.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+const rule = require('unified-lint-rule');
+const visit = require('unist-util-visit');
+const titleCase = require('to-title-case');
```

Comment:
[`to-title-case`](https://github.com/ianstormtaylor/to-title-case) seems like the best option here since it was made by someone who seems to be reputable and active, and is properly tested (at least from how it seemed to me).

Reply:
`to-title-case` looks fine, but see my other comment below regarding handling other casings aside from title case. since we're already using `case` for `list-item`, i'd recommend that we use that instead of bringing in another dependency.